astro_2.0/features/pages/cfcLibrary/hooks.ts

```diff
@@ -0,0 +1,143 @@
+import { useCallback, useEffect, useState } from 'react';
+import { useAsyncFn, useMountedState } from 'react-use';
+import { SputnikHttpService } from 'services/sputnik';
+import { LIST_LIMIT_DEFAULT } from 'services/sputnik/constants';
+import { PaginationResponse } from 'types/api';
+import { SharedProposalTemplate } from 'types/proposalTemplate';
+import { useRouter } from 'next/router';
+import { useDebounceEffect } from 'hooks/useDebounceUpdateEffect';
+
+export type PageData = PaginationResponse<SharedProposalTemplate[]>;
+
+export function useCfcLibraryData(): {
```

Comment:
Can we split that into 2 separate files - `useCfcLibraryData` and `useSharedTemplatePageData` and place them into `hooks` folder?

Reply:
I would like to keep these 2 hook at the same place. They are now application wide used, they are very special to specific page and defined to keep page logic only.